realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java

```diff
@@ -270,4 +303,30 @@ public int hashCode() {
         return 37;
     }
 
+
+    // Helper class for keeping track of strong references to objects.
+    private static class StrongReferenceCounter<K> {
+
+        private final Map<K, Integer> references = new IdentityHashMap<K, Integer>();
```

Comment:
But that is the point of the IdentityHashMap?

It uses reference equality (`==`) instead of `equals` to avoid the exact scenario you are describing.